Subsystems/nosSettingsSubsystem/Include/nosSettingsSubsystem/nosSettingsSubsystem.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright MediaZ Teknoloji A.S. All Rights Reserved.
+ */
+
+#ifndef NOS_SETTINGS_SUBSYSTEM_H_INCLUDED
+#define NOS_SETTINGS_SUBSYSTEM_H_INCLUDED
+#include "Nodos/Types.h"
+
+typedef enum nosSettingsFileDirectory {
+	NOS_SETTINGS_FILE_DIRECTORY_LOCAL, // Module's root folder
+	NOS_SETTINGS_FILE_DIRECTORY_WORKSPACE, // Engine's binary folder
+	NOS_SETTINGS_FILE_DIRECTORY_GLOBAL	// AppData folder
+} nosSettingsFileDirectory;
+
+typedef void(*nosPfnSettingsItemUpdate)(nosName itemName, nosBuffer itemValue);
+
+#if NOS_HAS_CPLUSPLUS_20
+#include "EditorEvents_generated.h"
+typedef nos::sys::settings::editor::SettingsList nosSettingsList;
+#else
+typedef void* nosSettingsList;
+#endif
+
+struct nosSettingsSubsystem
+{
+	nosResult(__stdcall* ReadSettings)(nosName typeName, nosBuffer* buffer);
+	nosResult(__stdcall* WriteSettings)(nosName typeName, nosBuffer buffer, nosSettingsFileDirectory directory);
```

Comment:
We should also add a nosName key parameter at the start of this. For example, in the current API, it is not possible for a module to write two 'float's to the settings.

Reply:
To support module settings that only contains a single entry, we could also add default read/write function pairs like this, which adds a single entry to the settings with key "default".